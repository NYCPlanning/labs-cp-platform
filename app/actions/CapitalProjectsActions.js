import dispatcher from '../dispatcher';

const CapitalProjectsActions = {
  onFilterDimensionChange(filterDimension, values) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_FILTERDIMENSION_CHANGE',
      filterDimension,
      values,
    });
  },

  fetchDetailData(id) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_FETCH_DETAIL_DATA',
      id,
    });
  },

  resetFilter() {
    dispatcher.dispatch({
      type: 'CAPTIALPROJECTS_RESET_FILTER',
    });
  },

  setSelectedFeatures(features) {
    dispatcher.dispatch({
      type: 'CAPTIALPROJECTS_SET_SELECTED_FEATURES',
      features,
    });
  },
};

export default CapitalProjectsActions;
