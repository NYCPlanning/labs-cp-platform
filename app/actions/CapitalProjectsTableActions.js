import dispatcher from '../dispatcher';

const CapitalProjectsTableActions = {
  onFilterDimensionChange(filterDimension, values) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_TABLE_FILTERDIMENSION_CHANGE',
      filterDimension,
      values,
    });
  },

  onFilterBy(filterBy) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_TABLE_FILTERBY',
      filterBy,
    });
  },

  onSortChange(columnKey, sortDir) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_TABLE_SORTCHANGE',
      columnKey,
      sortDir,
    });
  },
};

export default CapitalProjectsTableActions;
