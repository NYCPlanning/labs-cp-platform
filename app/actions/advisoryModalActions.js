import * as AT from '../constants/actionTypes';

export const hideModal = () => ({
  type: AT.HIDE_MODAL,
});

export const showModal = () => ({
  type: AT.SHOW_MODAL,
});
