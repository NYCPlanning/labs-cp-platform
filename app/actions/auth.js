import * as AT from '../constants/actionTypes';

export const login = ({ params }) => ({
  type: AT.AUTH0_LOGIN,
  payload: { params },
});

export const logout = () => ({
  type: AT.AUTH0_LOGOUT,
});

export const signUp = () => ({
  type: AT.AUTH0_SIGNUP,
});

export const loadCredentials = ({ targetPath }) => ({
  type: AT.LOAD_CREDENTIALS,
  payload: { targetPath },
});

export const authorizeUser = (profile, token) => ({
  type: AT.AUTHORIZE_USER,
  payload: { profile, token },
});

export const deauthorizeUser = () => ({
  type: AT.DEAUTHORIZE_USER,
});

export const ensureAccess = (permission) => ({
  type: AT.ENSURE_ACCESS,
});
