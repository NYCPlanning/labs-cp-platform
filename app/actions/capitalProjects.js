import * as AT from '../constants/actionTypes';
import * as cartoActions from './carto';
import { sqlConfig, unionSql } from '../helpers/sqlbuilder/CapitalProjectsSqlBuilder';

export const fetchDetails = capitalProjectId =>
  cartoActions.getFeature({
    tableName: sqlConfig.combinedTable,
    column: 'maprojid',
    value: capitalProjectId,
  }, AT.FETCH_CAPITAL_PROJECT_DETAILS);

export const fetchBudgets = capitalProjectId => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT * FROM cpdb_budgets WHERE maprojid = '${capitalProjectId}'`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECT_BUDGETS,
  },
});

export const fetchCommitments = capitalProjectId => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT * FROM cpdb_commitments WHERE maprojid = '${capitalProjectId}' ORDER BY to_date(plancommdate,'MM/YY')`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECT_COMMITMENTS,
  },
});

export const fetchTotalPointsCount = () => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*) FROM ${sqlConfig.pointsTablename}`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_TOTAL_POINTS_COUNT,
  },
});

export const fetchTotalPolygonsCount = () => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*) FROM ${sqlConfig.polygonsTablename}`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_TOTAL_POLYGONS_COUNT,
  },
});

export const fetchSelectedCount = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT count(*) FROM (${unionSql(filterDimensions)}) a`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_SELECTED_COUNT,
  },
});

export const setSelectedFeatures = selectedFeatures => ({
  type: AT.SET_SELECTED_CAPITAL_PROJECTS_FEATURES,
  payload: { selectedFeatures },
});

export const setFilterDimension = (filterDimension, values) => ({
  type: AT.SET_CAPITAL_PROJECTS_FILTER_DIMENSION,
  payload: { filterDimension, values },
});

export const resetFilters = () => ({
  type: AT.RESET_CAPITAL_PROJECTS_FILTERS,
});
