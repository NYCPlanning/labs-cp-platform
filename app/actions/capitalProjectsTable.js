import * as AT from '../constants/actionTypes';
import { getTableSql } from '../helpers/sqlbuilder/CapitalProjectsSqlBuilder';

export const fetchDetails = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: getTableSql(filterDimensions),
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECT_TABLE_DETAILS,
  },
});

export const fetchSelectedCount = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*) FROM (${getTableSql(filterDimensions)}) a`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_TABLE_SELECTED_COUNT,
  },
});

export const setFilterDimension = (filterDimension, values) => ({
  type: AT.SET_CAPITAL_PROJECTS_TABLE_FILTER_DIMENSION,
  payload: { filterDimension, values },
});

export const resetFilter = () => ({
  type: AT.RESET_CAPITAL_PROJECTS_TABLE_FILTER,
});

export const setTableSort = (columnKey, sortDir) => ({
  type: AT.SET_CAPITAL_PROJECTS_TABLE_SORT,
  payload: { columnKey, sortDir },
});

export const setTableFilterBy = filterBy => ({
  type: AT.SET_CAPITAL_PROJECTS_TABLE_FILTER_BY,
  payload: { filterBy },
});
