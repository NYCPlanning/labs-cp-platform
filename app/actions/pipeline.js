import * as AT from '../constants/actionTypes';
import * as cartoActions from './carto';
import { sqlConfig, getSql } from '../helpers/sqlbuilder/PipelineSqlBuilder';

export const fetchDetails = cartodbId =>
  cartoActions.getFeature({
    tableName: sqlConfig.tablename,
    column: 'cartodb_id',
    value: cartodbId,
  }, AT.FETCH_PIPELINE_DETAILS);

export const fetchTotalCount = () => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*)
    FROM ${sqlConfig.tablename}
    WHERE
      the_geom IS NOT NULL
      AND
      u_net IS NOT NULL
      AND
      (
        dcp_status = 'Complete'
        OR dcp_status = 'Partial complete'
        OR dcp_status = 'Permit issued'
        OR dcp_status = 'Application filed'
        OR dcp_status = 'Complete (demolition)'
      )
      AND
      (
        dcp_dev_category = 'New Building'
        OR dcp_dev_category = 'Alteration'
        OR dcp_dev_category = 'Demolition'
      )
      AND
      (
        dcp_occ_category = 'Residential'
        OR dcp_occ_category = 'Other Accommodations'
      )
      AND x_outlier <> 'true'
      AND x_dup_flag = ''`,
    requestFormat: 'json',
    nextType: AT.FETCH_PIPELINE_TOTAL_COUNT,
  },
});

export const fetchSelectedCount = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT count(*) FROM (${getSql(filterDimensions)}) a`,
    requestFormat: 'json',
    nextType: AT.FETCH_PIPELINE_SELECTED_COUNT,
  },
});

export const setSelectedFeatures = selectedFeatures => ({
  type: AT.SET_SELECTED_PIPELINE_FEATURES,
  payload: { selectedFeatures },
});

export const setFilterDimension = (filterDimension, values) => ({
  type: AT.SET_PIPELINE_FILTER_DIMENSION,
  payload: { filterDimension, values },
});

export const setSymbology = symbologyDimension => ({
  type: AT.SET_PIPELINE_SYMBOLOGY,
  payload: { symbologyDimension },
});

export const resetFilter = () => ({
  type: AT.RESET_PIPELINE_FILTER,
});
