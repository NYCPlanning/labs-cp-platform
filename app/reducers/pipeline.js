import * as AT from '../constants/actionTypes';
import { defaultFilterDimensions } from '../pipeline/config';
import { getSql } from '../helpers/sqlbuilder/PipelineSqlBuilder';
import _ from 'lodash';

const getDefaultFilters = () => JSON.parse(JSON.stringify(defaultFilterDimensions));

const isIssueDateDisabled = filterDimensions =>
filterDimensions.dcp_pipeline_status.values
  .filter(value => value.checked && value.value === 'Application filed')
  .length > 0;

const isCompletionDateDisabled = filterDimensions =>
filterDimensions.dcp_pipeline_status.values
  .filter(value => value.checked && (value.value === 'Permit issued' || value.value === 'Application filed'))
  .length > 0;

export const initialState = {
  filterDimensions: getDefaultFilters(),
  issueDateFilterDisabled: isIssueDateDisabled(getDefaultFilters()),
  completionDateFilterDisabled: isCompletionDateDisabled(getDefaultFilters()),
  sql: getSql(getDefaultFilters()),
  selectedFeatures: [],
  symbologyDimension: 'dcp_permit_type',
  pipelineDetails: null,
  totalCount: 0,
  selectedCount: 0,
};

const pipelineReducer = (state = initialState, action) => {
  switch (action.type) {

    case AT.FETCH_PIPELINE_DETAILS.SUCCESS:
      return Object.assign({}, state, { pipelineDetails: action.payload.features[0] });

    case AT.FETCH_PIPELINE_TOTAL_COUNT.SUCCESS:
      return Object.assign({}, state, { totalCount: action.payload[0].count });

    case AT.FETCH_PIPELINE_SELECTED_COUNT.SUCCESS:
      return Object.assign({}, state, { selectedCount: action.payload[0].count });

    case AT.SET_SELECTED_PIPELINE_FEATURES:
      return Object.assign({}, state, { selectedFeatures: action.payload.selectedFeatures });

    case AT.SET_PIPELINE_SYMBOLOGY:
      return Object.assign({}, state, { symbologyDimension: action.payload.symbologyDimension });

    case AT.RESET_PIPELINE_FILTER:
      return Object.assign({}, state, {
        filterDimensions: getDefaultFilters(),
        issueDateFilterDisabled: initialState.issueDateFilterDisabled,
        completionDateFilterDisabled: initialState.completionDateFilterDisabled,
        sql: getSql(getDefaultFilters()),
      });

    case AT.SET_PIPELINE_FILTER_DIMENSION:
      const { filterDimension, values } = action.payload;
      const dimensions = _.cloneDeep(state.filterDimensions);

      dimensions[filterDimension].values = values;

      // if dimension is status, check which items are included and disable/reset date dimensions accordingly
      if (filterDimension === 'dcp_pipeline_status') {
        // Completion Slider
        if (isCompletionDateDisabled(dimensions)) {
          dimensions.dob_cofo_date = getDefaultFilters().dob_cofo_date;
          dimensions.dob_cofo_date.disabled = true;
        } else {
          dimensions.dob_cofo_date.disabled = false;
        }
        // issued slider
        if (isIssueDateDisabled(dimensions)) {
          dimensions.dob_qdate = getDefaultFilters().dob_qdate;
          dimensions.dob_qdate.disabled = true;
        } else {
          dimensions.dob_qdate.disabled = false;
        }
      }

      return Object.assign({}, state, {
        filterDimensions: dimensions,
        issueDateFilterDisabled: isIssueDateDisabled(dimensions),
        completionDateFilterDisabled: isCompletionDateDisabled(dimensions),
        sql: getSql(dimensions)
      });

    default:
      return state;
  }
};

export default pipelineReducer;
