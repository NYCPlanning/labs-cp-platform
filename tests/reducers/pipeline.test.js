import _ from 'lodash';
import * as actions from '../../app/actions/pipeline';
import pipelineReducer, { initialState } from '../../app/reducers/pipeline';

describe('reducers/pipeline', () => {
  test('AT.FETCH_PIPELINE_DETAILS.SUCCESS', () =>
    expect(pipelineReducer(initialState, actions.fetchDetails('cartodbId'))).toMatchSnapshot());

  test('AT.FETCH_PIPELINE_TOTAL_COUNT.SUCCESS', () =>
    expect(pipelineReducer(initialState, actions.fetchTotalCount())).toMatchSnapshot());

  test('AT.FETCH_PIPELINE_SELECTED_COUNT.SUCCESS', () =>
    expect(pipelineReducer(initialState, actions.fetchSelectedCount(defaultFilterDimensions))).toMatchSnapshot());

  test('AT.SET_SELECTED_PIPELINE_FEATURES', () =>
    expect(pipelineReducer(initialState, actions.setSelectedFeatures([]))).toMatchSnapshot());

  test('AT.SET_PIPELINE_SYMBOLOGY', () =>
    expect(pipelineReducer(initialState, actions.setSymbology('symbologyDimension'))).toMatchSnapshot());

  test('AT.RESET_PIPELINE_FILTERS', () =>
    expect(pipelineReducer(initialState, actions.resetFilters())).toMatchSnapshot());

  test('AT.SET_PIPELINE_FILTER_DIMENSION', () =>
    expect(pipelineReducer(initialState, actions.setFilterDimension('dimension', 'values'))).toMatchSnapshot());

  test('AT.SET_PIPELINE_FILTER_DIMENSION - dcp_pipeline_status - completion date disabled', () => {
    const state = _.assign({}, initialState, {
      filterDimensions: _.assign({}, initialState.filterDimensions, {
        dcp_pipeline_status: { values: [{ checked: true, value: 'Application filed' }] }
      })
    });

    expect(pipelineReducer(state, actions.setFilterDimension('dcp_pipeline_status', 'values'))).toMatchSnapshot();
  });

  test('AT.SET_PIPELINE_FILTER_DIMENSION - dcp_pipeline_status - completion date enabled', () => {
    const state = _.assign({}, initialState, {
      filterDimensions: _.assign({}, initialState.filterDimensions, {
        dcp_pipeline_status: { values: [{ checked: false, value: 'Application filed' }] }
      })
    });

    expect(pipelineReducer(state, actions.setFilterDimension('dcp_pipeline_status', 'values'))).toMatchSnapshot()
  });

  test('AT.SET_PIPELINE_FILTER_DIMENSION - dcp_pipeline_status - issue date disabled', () => {
    const state = _.assign({}, initialState, {
      filterDimensions: _.assign({}, initialState.filterDimensions, {
        dcp_pipeline_status: { values: [{ checked: true, value: 'Permit issued' }] }
      })
    });

    expect(pipelineReducer(state, actions.setFilterDimension('dcp_pipeline_status', 'values'))).toMatchSnapshot()
  });

  test('AT.SET_PIPELINE_FILTER_DIMENSION - dcp_pipeline_status - issue date enabled', () => {
    const state = _.assign({}, initialState, {
      filterDimensions: _.assign({}, initialState.filterDimensions, {
        dcp_pipeline_status: { values: [{ checked: false, value: 'Permit issued' }] }
      })
    });

    expect(pipelineReducer(state, actions.setFilterDimension('dcp_pipeline_status', 'values'))).toMatchSnapshot()
  });
});
