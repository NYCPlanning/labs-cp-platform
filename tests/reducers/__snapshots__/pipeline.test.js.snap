// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/modalReducer CLOSE_MODAL 1`] = `null`;

exports[`reducers/modalReducer CLOSE_MODAL 2`] = `null`;

exports[`reducers/modalReducer OPEN_MODAL - reopen existing modal 1`] = `
Object {
  "modalCloseText": "New Close Text",
  "modalContent": "New Modal Content",
  "modalHeading": "New Modal Heading",
}
`;

exports[`reducers/modalReducer OPEN_MODAL 1`] = `
Object {
  "modalCloseText": "Close Text",
  "modalContent": "Modal Content",
  "modalHeading": "Modal Heading",
}
`;

exports[`reducers/pipeline AT.FETCH_PIPELINE_DETAILS.SUCCESS 1`] = `
Object {
  "completionDateFilterDisabled": true,
  "filterDimensions": Object {
    "dcp_development_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "label": "Residential",
          "value": "Residential",
        },
        Object {
          "checked": true,
          "label": "Non-residential",
          "value": "Non-residential",
        },
      ],
    },
    "dcp_permit_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "rgba(0, 228, 14, 0.7)",
          "label": "New Building",
          "value": "New Building",
        },
        Object {
          "checked": true,
          "color": "rgba(81, 99, 230, 0.77)",
          "label": "Alteration",
          "value": "Alteration",
        },
        Object {
          "checked": true,
          "color": "rgba(234, 62, 62, 1)",
          "label": "Demolition",
          "value": "Demolition",
        },
      ],
    },
    "dcp_pipeline_status": Object {
      "type": "statusMultiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "#238b45",
          "label": "Complete",
          "value": "Complete",
        },
        Object {
          "checked": true,
          "color": "#74c476",
          "label": "Partial complete",
          "value": "Partial complete",
        },
        Object {
          "checked": true,
          "color": "#bae4b3",
          "label": "Permit issued",
          "value": "Permit issued",
        },
        Object {
          "checked": true,
          "color": "#edf8e9",
          "label": "Application filed",
          "value": "Application filed",
        },
      ],
    },
    "dcp_units_use_map": Object {
      "type": "numberRange",
      "values": Array [
        -1445,
        1669,
      ],
    },
    "dob_cofo_date": Object {
      "disabled": true,
      "type": "cofoDateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
    "dob_qdate": Object {
      "disabled": true,
      "type": "dateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
  },
  "issueDateFilterDisabled": true,
  "pipelineDetails": null,
  "selectedCount": 0,
  "selectedFeatures": Array [],
  "sql": "SELECT cartodb_id, the_geom_webmercator, dcp_pipeline_status, dcp_permit_type, dcp_units_use_map, dob_permit_address FROM pipeline_projects_prod WHERE (dcp_pipeline_status = 'Complete' OR dcp_pipeline_status = 'Partial complete' OR dcp_pipeline_status = 'Permit issued' OR dcp_pipeline_status = 'Application filed' OR dcp_pipeline_status = 'Demolition (complete)') AND (dcp_permit_type = 'New Building' OR dcp_permit_type = 'Alteration' OR dcp_permit_type = 'Demolition') AND (dcp_development_type = 'Residential' OR dcp_development_type = 'Non-residential') AND (dcp_units_use_map >= '-1445' AND dcp_units_use_map <= '1669')",
  "symbologyDimension": "dcp_permit_type",
  "totalCount": 0,
}
`;

exports[`reducers/pipeline AT.FETCH_PIPELINE_TOTAL_COUNT.SUCCESS 1`] = `
Object {
  "completionDateFilterDisabled": true,
  "filterDimensions": Object {
    "dcp_development_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "label": "Residential",
          "value": "Residential",
        },
        Object {
          "checked": true,
          "label": "Non-residential",
          "value": "Non-residential",
        },
      ],
    },
    "dcp_permit_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "rgba(0, 228, 14, 0.7)",
          "label": "New Building",
          "value": "New Building",
        },
        Object {
          "checked": true,
          "color": "rgba(81, 99, 230, 0.77)",
          "label": "Alteration",
          "value": "Alteration",
        },
        Object {
          "checked": true,
          "color": "rgba(234, 62, 62, 1)",
          "label": "Demolition",
          "value": "Demolition",
        },
      ],
    },
    "dcp_pipeline_status": Object {
      "type": "statusMultiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "#238b45",
          "label": "Complete",
          "value": "Complete",
        },
        Object {
          "checked": true,
          "color": "#74c476",
          "label": "Partial complete",
          "value": "Partial complete",
        },
        Object {
          "checked": true,
          "color": "#bae4b3",
          "label": "Permit issued",
          "value": "Permit issued",
        },
        Object {
          "checked": true,
          "color": "#edf8e9",
          "label": "Application filed",
          "value": "Application filed",
        },
      ],
    },
    "dcp_units_use_map": Object {
      "type": "numberRange",
      "values": Array [
        -1445,
        1669,
      ],
    },
    "dob_cofo_date": Object {
      "disabled": true,
      "type": "cofoDateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
    "dob_qdate": Object {
      "disabled": true,
      "type": "dateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
  },
  "issueDateFilterDisabled": true,
  "pipelineDetails": null,
  "selectedCount": 0,
  "selectedFeatures": Array [],
  "sql": "SELECT cartodb_id, the_geom_webmercator, dcp_pipeline_status, dcp_permit_type, dcp_units_use_map, dob_permit_address FROM pipeline_projects_prod WHERE (dcp_pipeline_status = 'Complete' OR dcp_pipeline_status = 'Partial complete' OR dcp_pipeline_status = 'Permit issued' OR dcp_pipeline_status = 'Application filed' OR dcp_pipeline_status = 'Demolition (complete)') AND (dcp_permit_type = 'New Building' OR dcp_permit_type = 'Alteration' OR dcp_permit_type = 'Demolition') AND (dcp_development_type = 'Residential' OR dcp_development_type = 'Non-residential') AND (dcp_units_use_map >= '-1445' AND dcp_units_use_map <= '1669')",
  "symbologyDimension": "dcp_permit_type",
  "totalCount": 0,
}
`;

exports[`reducers/pipeline AT.RESET_PIPELINE_FILTERS 1`] = `
Object {
  "completionDateFilterDisabled": true,
  "filterDimensions": Object {
    "dcp_development_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "label": "Residential",
          "value": "Residential",
        },
        Object {
          "checked": true,
          "label": "Non-residential",
          "value": "Non-residential",
        },
      ],
    },
    "dcp_permit_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "rgba(0, 228, 14, 0.7)",
          "label": "New Building",
          "value": "New Building",
        },
        Object {
          "checked": true,
          "color": "rgba(81, 99, 230, 0.77)",
          "label": "Alteration",
          "value": "Alteration",
        },
        Object {
          "checked": true,
          "color": "rgba(234, 62, 62, 1)",
          "label": "Demolition",
          "value": "Demolition",
        },
      ],
    },
    "dcp_pipeline_status": Object {
      "type": "statusMultiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "#238b45",
          "label": "Complete",
          "value": "Complete",
        },
        Object {
          "checked": true,
          "color": "#74c476",
          "label": "Partial complete",
          "value": "Partial complete",
        },
        Object {
          "checked": true,
          "color": "#bae4b3",
          "label": "Permit issued",
          "value": "Permit issued",
        },
        Object {
          "checked": true,
          "color": "#edf8e9",
          "label": "Application filed",
          "value": "Application filed",
        },
      ],
    },
    "dcp_units_use_map": Object {
      "type": "numberRange",
      "values": Array [
        -1445,
        1669,
      ],
    },
    "dob_cofo_date": Object {
      "disabled": true,
      "type": "cofoDateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
    "dob_qdate": Object {
      "disabled": true,
      "type": "dateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
  },
  "issueDateFilterDisabled": true,
  "pipelineDetails": null,
  "selectedCount": 0,
  "selectedFeatures": Array [],
  "sql": "SELECT cartodb_id, the_geom_webmercator, dcp_pipeline_status, dcp_permit_type, dcp_units_use_map, dob_permit_address FROM pipeline_projects_prod WHERE (dcp_pipeline_status = 'Complete' OR dcp_pipeline_status = 'Partial complete' OR dcp_pipeline_status = 'Permit issued' OR dcp_pipeline_status = 'Application filed' OR dcp_pipeline_status = 'Demolition (complete)') AND (dcp_permit_type = 'New Building' OR dcp_permit_type = 'Alteration' OR dcp_permit_type = 'Demolition') AND (dcp_development_type = 'Residential' OR dcp_development_type = 'Non-residential') AND (dcp_units_use_map >= '-1445' AND dcp_units_use_map <= '1669')",
  "symbologyDimension": "dcp_permit_type",
  "totalCount": 0,
}
`;

exports[`reducers/pipeline AT.SET_PIPELINE_SYMBOLOGY 1`] = `
Object {
  "completionDateFilterDisabled": true,
  "filterDimensions": Object {
    "dcp_development_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "label": "Residential",
          "value": "Residential",
        },
        Object {
          "checked": true,
          "label": "Non-residential",
          "value": "Non-residential",
        },
      ],
    },
    "dcp_permit_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "rgba(0, 228, 14, 0.7)",
          "label": "New Building",
          "value": "New Building",
        },
        Object {
          "checked": true,
          "color": "rgba(81, 99, 230, 0.77)",
          "label": "Alteration",
          "value": "Alteration",
        },
        Object {
          "checked": true,
          "color": "rgba(234, 62, 62, 1)",
          "label": "Demolition",
          "value": "Demolition",
        },
      ],
    },
    "dcp_pipeline_status": Object {
      "type": "statusMultiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "#238b45",
          "label": "Complete",
          "value": "Complete",
        },
        Object {
          "checked": true,
          "color": "#74c476",
          "label": "Partial complete",
          "value": "Partial complete",
        },
        Object {
          "checked": true,
          "color": "#bae4b3",
          "label": "Permit issued",
          "value": "Permit issued",
        },
        Object {
          "checked": true,
          "color": "#edf8e9",
          "label": "Application filed",
          "value": "Application filed",
        },
      ],
    },
    "dcp_units_use_map": Object {
      "type": "numberRange",
      "values": Array [
        -1445,
        1669,
      ],
    },
    "dob_cofo_date": Object {
      "disabled": true,
      "type": "cofoDateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
    "dob_qdate": Object {
      "disabled": true,
      "type": "dateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
  },
  "issueDateFilterDisabled": true,
  "pipelineDetails": null,
  "selectedCount": 0,
  "selectedFeatures": Array [],
  "sql": "SELECT cartodb_id, the_geom_webmercator, dcp_pipeline_status, dcp_permit_type, dcp_units_use_map, dob_permit_address FROM pipeline_projects_prod WHERE (dcp_pipeline_status = 'Complete' OR dcp_pipeline_status = 'Partial complete' OR dcp_pipeline_status = 'Permit issued' OR dcp_pipeline_status = 'Application filed' OR dcp_pipeline_status = 'Demolition (complete)') AND (dcp_permit_type = 'New Building' OR dcp_permit_type = 'Alteration' OR dcp_permit_type = 'Demolition') AND (dcp_development_type = 'Residential' OR dcp_development_type = 'Non-residential') AND (dcp_units_use_map >= '-1445' AND dcp_units_use_map <= '1669')",
  "symbologyDimension": "symbologyDimension",
  "totalCount": 0,
}
`;

exports[`reducers/pipeline AT.SET_SELECTED_PIPELINE_FEATURES 1`] = `
Object {
  "completionDateFilterDisabled": true,
  "filterDimensions": Object {
    "dcp_development_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "label": "Residential",
          "value": "Residential",
        },
        Object {
          "checked": true,
          "label": "Non-residential",
          "value": "Non-residential",
        },
      ],
    },
    "dcp_permit_type": Object {
      "type": "multiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "rgba(0, 228, 14, 0.7)",
          "label": "New Building",
          "value": "New Building",
        },
        Object {
          "checked": true,
          "color": "rgba(81, 99, 230, 0.77)",
          "label": "Alteration",
          "value": "Alteration",
        },
        Object {
          "checked": true,
          "color": "rgba(234, 62, 62, 1)",
          "label": "Demolition",
          "value": "Demolition",
        },
      ],
    },
    "dcp_pipeline_status": Object {
      "type": "statusMultiSelect",
      "values": Array [
        Object {
          "checked": true,
          "color": "#238b45",
          "label": "Complete",
          "value": "Complete",
        },
        Object {
          "checked": true,
          "color": "#74c476",
          "label": "Partial complete",
          "value": "Partial complete",
        },
        Object {
          "checked": true,
          "color": "#bae4b3",
          "label": "Permit issued",
          "value": "Permit issued",
        },
        Object {
          "checked": true,
          "color": "#edf8e9",
          "label": "Application filed",
          "value": "Application filed",
        },
      ],
    },
    "dcp_units_use_map": Object {
      "type": "numberRange",
      "values": Array [
        -1445,
        1669,
      ],
    },
    "dob_cofo_date": Object {
      "disabled": true,
      "type": "cofoDateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
    "dob_qdate": Object {
      "disabled": true,
      "type": "dateRange",
      "values": Array [
        "1293840000",
        "1497905065",
      ],
    },
  },
  "issueDateFilterDisabled": true,
  "pipelineDetails": null,
  "selectedCount": 0,
  "selectedFeatures": Array [],
  "sql": "SELECT cartodb_id, the_geom_webmercator, dcp_pipeline_status, dcp_permit_type, dcp_units_use_map, dob_permit_address FROM pipeline_projects_prod WHERE (dcp_pipeline_status = 'Complete' OR dcp_pipeline_status = 'Partial complete' OR dcp_pipeline_status = 'Permit issued' OR dcp_pipeline_status = 'Application filed' OR dcp_pipeline_status = 'Demolition (complete)') AND (dcp_permit_type = 'New Building' OR dcp_permit_type = 'Alteration' OR dcp_permit_type = 'Demolition') AND (dcp_development_type = 'Residential' OR dcp_development_type = 'Non-residential') AND (dcp_units_use_map >= '-1445' AND dcp_units_use_map <= '1669')",
  "symbologyDimension": "dcp_permit_type",
  "totalCount": 0,
}
`;
